conda_smithy/linter/hints.py

```diff
@@ -291,24 +293,74 @@ def hint_noarch_python_use_python_min(
                 ):
                     break
             else:
-                hint += (
-                    f"\n   - For the `{section_name}` section of the recipe, you should usually use `{report_syntax}` "
+                section_desc = (
+                    f"`{output_name}` output" if output_name else "recipe"
+                )
+                hint.append(
+                    f"\n   - For the `{section_name}` section of {section_desc}, you should usually use `{report_syntax}` "
                     f"for the `python` entry."
                 )
+    return hint
 
-        if hint:
-            hint = (
-                (
-                    "`noarch: python` recipes should usually follow the syntax in "
-                    "our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) "
-                    "for specifying the Python version."
-                )
-                + hint
-                + (
-                    "\n   - If the package requires a newer Python version than the currently supported minimum "
-                    "version on `conda-forge`, you can override the `python_min` variable by adding a "
-                    "Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` "
-                    "variable for v1 recipes)."
+
+def hint_noarch_python_use_python_min(
+    host_reqs,
+    run_reqs,
+    test_reqs,
+    outputs_section,
+    noarch_value,
+    recipe_version,
+    hints,
+):
+    hint = []
+
+    if outputs_section:
+        for output_num, output in enumerate(outputs_section):
+            requirements = output.get("requirements", {})
+            if isinstance(requirements, Mapping):
+                output_host_reqs = requirements.get("host")
+                output_run_reqs = requirements.get("run")
+            else:
+                output_host_reqs = requirements
```

Comment:
```suggestion
                output_host_reqs = None
```

Reply:
Requirements that are not in a mapping are always run for outputs of v0 recipes.